File: qc-core/src/main/java/com/eager/questioncloud/core/domain/feed/library/service/LibraryService.java
package com.eager.questioncloud.core.domain.feed.library.service;

import com.eager.questioncloud.core.domain.feed.library.dto.UserQuestionDto.UserQuestionItem;
import com.eager.questioncloud.core.domain.feed.library.implement.UserQuestionReader;
import com.eager.questioncloud.core.domain.hub.question.common.QuestionFilter;
import java.util.List;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class LibraryService {
    private final UserQuestionReader userQuestionReader;

    public List<UserQuestionItem> getUserQuestions(QuestionFilter questionFilter) {
        return userQuestionReader.getUserQuestions(questionFilter);
    }

    public int countUserQuestions(QuestionFilter questionFilter) {
        return userQuestionReader.countUserQuestions(questionFilter);
    }
}

File: qc-api/qc-library-api/src/main/kotlin/com/eager/questioncloud/library/service/LibraryService.kt
package com.eager.questioncloud.library.service

import com.eager.questioncloud.library.dto.LibraryContent
import com.eager.questioncloud.library.implement.LibraryContentReader
import com.eager.questioncloud.question.common.QuestionFilter
import org.springframework.stereotype.Service

@Service
class LibraryService(
    private val libraryContentReader: LibraryContentReader
) {
    fun getUserQuestions(questionFilter: QuestionFilter): List<LibraryContent> {
        return libraryContentReader.getUserQuestions(questionFilter)
    }

    fun countUserQuestions(questionFilter: QuestionFilter): Int {
        return libraryContentReader.countUserQuestions(questionFilter)
    }
}
